scene/gui/virtual_joystick.cpp

```diff
@@ -0,0 +1,381 @@
+
+	ClassDB::bind_method(D_METHOD("set_clampzone_size", "size"), &VirtualJoystick::set_clampzone_size);
+	ClassDB::bind_method(D_METHOD("get_clampzone_size"), &VirtualJoystick::get_clampzone_size);
+
+	ClassDB::bind_method(D_METHOD("set_initial_offset_ratio", "ratio"), &VirtualJoystick::set_initial_offset_ratio);
+	ClassDB::bind_method(D_METHOD("get_initial_offset_ratio"), &VirtualJoystick::get_initial_offset_ratio);
+
+	ClassDB::bind_method(D_METHOD("set_action_left", "action"), &VirtualJoystick::set_action_left);
+	ClassDB::bind_method(D_METHOD("get_action_left"), &VirtualJoystick::get_action_left);
+
+	ClassDB::bind_method(D_METHOD("set_action_right", "action"), &VirtualJoystick::set_action_right);
+	ClassDB::bind_method(D_METHOD("get_action_right"), &VirtualJoystick::get_action_right);
+
+	ClassDB::bind_method(D_METHOD("set_action_up", "action"), &VirtualJoystick::set_action_up);
+	ClassDB::bind_method(D_METHOD("get_action_up"), &VirtualJoystick::get_action_up);
+
+	ClassDB::bind_method(D_METHOD("set_action_down", "action"), &VirtualJoystick::set_action_down);
+	ClassDB::bind_method(D_METHOD("get_action_down"), &VirtualJoystick::get_action_down);
+
+	ClassDB::bind_method(D_METHOD("set_visibility_mode", "mode"), &VirtualJoystick::set_visibility_mode);
+	ClassDB::bind_method(D_METHOD("get_visibility_mode"), &VirtualJoystick::get_visibility_mode);
+
+	ClassDB::bind_method(D_METHOD("set_joystick_texture", "texture"), &VirtualJoystick::set_joystick_texture);
+	ClassDB::bind_method(D_METHOD("get_joystick_texture"), &VirtualJoystick::get_joystick_texture);
+	ClassDB::bind_method(D_METHOD("set_tip_texture", "texture"), &VirtualJoystick::set_tip_texture);
+	ClassDB::bind_method(D_METHOD("get_tip_texture"), &VirtualJoystick::get_tip_texture);
+
+	ADD_SIGNAL(MethodInfo("pressed"));
+	ADD_SIGNAL(MethodInfo("tapped"));
+	ADD_SIGNAL(MethodInfo("released", PropertyInfo(Variant::VECTOR2, "input_vector")));
+	ADD_SIGNAL(MethodInfo("flicked", PropertyInfo(Variant::VECTOR2, "input_vector")));
+	ADD_SIGNAL(MethodInfo("flick_canceled"));
+
+	ADD_PROPERTY(PropertyInfo(Variant::INT, "joystick_mode", PROPERTY_HINT_ENUM, "Fixed,Dynamic,Following"), "set_joystick_mode", "get_joystick_mode");
+
+	ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "joystick_size", PROPERTY_HINT_RANGE, "10,500,10"), "set_joystick_size", "get_joystick_size");
+	ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "deadzone_size", PROPERTY_HINT_RANGE, "0,500,5"), "set_deadzone_size", "get_deadzone_size");
+	ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "clampzone_size", PROPERTY_HINT_RANGE, "10,500,10"), "set_clampzone_size", "get_clampzone_size");
+	ADD_PROPERTY(PropertyInfo(Variant::VECTOR2, "initial_offset_ratio", PROPERTY_HINT_RANGE, "0.0,1,0.01"), "set_initial_offset_ratio", "get_initial_offset_ratio");
+
+	ADD_PROPERTY(PropertyInfo(Variant::STRING_NAME, "action_left", PROPERTY_HINT_INPUT_NAME), "set_action_left", "get_action_left");
```

Comment:
```suggestion
	ADD_PROPERTY(PropertyInfo(Variant::STRING_NAME, "action_left", PROPERTY_HINT_INPUT_NAME, "show_builtin,loose_mode"), "set_action_left", "get_action_left");
```
To match TouchScreenButton (and to show the default value 🙃).
Same below.

Reply:
Added